include/aspect/compat.h

```diff
@@ -83,6 +83,67 @@ namespace dealii
 }
 #endif
 
+/*
+ * Unique_ptr functionality that was introduced in deal.II 8.3 and replaces
+ * auto_ptr in CXX17. We need this to silence deprecation warnings in new
+ * compilers.
+ */
+#if !DEAL_II_VERSION_GTE(8,3,0)
```

Comment:
Someone may include it somewhere and it may conflict because he just copies the contents of that file here. I think this is right.